core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3124,7 +3124,9 @@ object Stream extends StreamLowPriority {
     now.flatMap(go)
   }
 
-  final class PartiallyAppliedFromEither[F[_]] {
+  private[fs2] final class PartiallyAppliedFromEither[F[_]](
+      private val dummy: Boolean
```

Comment:
What's this boolean for?

Reply:
Oh, to allow `AnyVal` use?

Reply:
Yes, just for that.
I believe a Boolean is the smallest object one can use. Probably the JIT would inline all that.

The pattern is copied as it is from the [**cats guidelines**](https://typelevel.org/cats/guidelines.html) _(with the only difference that they use a default value)_.

Reply:
Gotcha, thanks! Will wait for feedback from Fabio but this LGTM.